### ISSUE TITLE
Bracketed virtual postings balance against whole transaction

### ISSUE THREAD

**Issue by twrightsman** [ledger/ledger#1827]
I'm not sure if this is a bug but it seems that bracketed virtual postings are balanced in the context of the whole transaction instead of only against other virtual postings. See this example, which should not balance (if I understand virtual postings correctly):

```
2019-09-12 Test
  Assets:Bank:Checking  $-10.00
  Expenses:Food  $11.00
  [Budgets:Total]  $10.00
  [Budgets:Food]  $-11.00
```

```
$ ledger -f test.ledg bal
             $-10.00  Assets:Bank:Checking
              $-1.00  Budgets
             $-11.00    Food
              $10.00    Total
              $11.00  Expenses:Food
--------------------
                   0
```

From the manual (Section 5.8):
> If you want, you can state that virtual postings should balance **against one or more other
virtual postings** by using brackets (which look “harder”) rather than parentheses

**Comment by tysonliddell** [comment]
Not sure if this is relevant, but the `--real` option gives the following (unbalanced) output:

```
$ ledger -f test.ledg bal --real
             $-10.00  Assets:Bank:Checking
              $11.00  Expenses:Food
--------------------
               $1.00
```

**Comment by tbm** [comment]
Sounds like a bug to me and not a documentation issue.

@jwiegley maybe you can comment on how this was supposed to work.

**Comment by jwiegley** [comment]
Yes, bracketing simply causes them to participate in the standard logic for balancing transactions. There is nothing special about a "group of virtual postings".

**Comment by kayb01000010** [comment]
i guess the OP wants a feature where virtual accounts only balance with other virtual accounts in a posting. This seems neither a bug (as sire jwiegley mentioned above) nor a documentation issue. Maybe this should be closed or tagged as a feature request ?

**Comment by tysonliddell** [comment]
@kayb01000010 but surely if the virtual accounts don't balance with the other virtual accounts using the `--real` option would cause transactions to no longer balance.

**Comment by jwiegley** [comment]
That's a great point, @tysonliddell. I think you're right that virtual accounts should only balance with other virtual accounts.
